confutils.nim

```diff
@@ -1154,13 +1156,13 @@ proc loadImpl[C, SecondarySources](
      subCmdDiscriminator.defaultSubCmd != -1 and
      fieldCounters[subCmdDiscriminator.idx] == 0:
     let defaultCmd = subCmdDiscriminator.subCmds[subCmdDiscriminator.defaultSubCmd]
-    activateCmd(subCmdDiscriminator, defaultCmd)
+    result.activateCmd(subCmdDiscriminator, defaultCmd)
 
-  if secondarySources != nil:
+  if not isNil(secondarySources):  # Nim v2.0.10: `!= nil` broken in nimscript
```

Comment:
```nim
type SecondarySources = object

proc loadImpl[C, SecondarySources](
    Configuration: typedesc[C],
    secondarySourcesRef: ref SecondarySources,
    secondarySources: proc (
        config: Configuration, sources: ref SecondarySources
    ) {.gcsafe.} = nil): Configuration =
  if secondarySources != nil:
    secondarySources(result, secondarySourcesRef)

template load(
    Configuration: type,
    secondarySources: untyped = nil): untyped =
  block:
    let secondarySourcesRef = new SecondarySources
    loadImpl(Configuration, secondarySourcesRef, secondarySources)

type
  CliConfig_33558379 = object
let config_33558380 = load(CliConfig_33558379)
```
is repro, and it's an issue in `v2.0.10` and `version-2-0` but not `version-2-2` or `devel`. Using `isNil` seems fine.

Reply:
it worked for me in `nim c` mode and only bugs out in `nim e` nimscript mode. 
do you have a link that I should refer to in the comment here?

Reply:
I don't, because it seems to already be fixed in Nim upstream, and it's not really worth a backport if it's just allowing `!= nil` instead of `not isNil`.

Can link to this comment in GitHub perhaps.